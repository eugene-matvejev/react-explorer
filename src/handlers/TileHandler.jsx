import React from 'react';
import Tile from '../components/Tile';
import HealthCheck from '../components/HealthCheck';

const fetchStatuses = callBack => callBack([
    {
      id: 1,
      status : "red",
      name : "Health Check 1",
      parent: null
    },
    {
      id: 2,
      status: "green",
      name: "Health Check 2",
      parent: null
    },
    {
      id: 3,
      status: "amber",
      name: "Health Check 3",
      parent: null
    }
]);

export default class TileHandler extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      statuses: [],
      isExpanded: false,
      isExpandedIndex: 0,
    }

    this.onClick = this.onClick.bind(this);
    this.handleClose = this.handleClose.bind(this);
  }

  //   componentDidMount() {
  //   const { onMount } = this.props;

  //   onMount && onMount(this.props, this.state, this.onSuccess, this.onError);

  //   fetchStatuses(statusArray => 
  //     this.setState({
  //       statuses: statusArray,
  //   }));
  // }

  setExpandedState(tileIndex) {
    const { isExpanded } = this.state;
    this.setState({
      isExpanded: !isExpanded,
      isExpandedIndex: tileIndex
    });
  }


  onClose(){
    this.setExpandedState(0);
  }

  onClick(e) {
    this.setExpandedState(tileId);
  }

  render() {
    const {statuses, isExpanded, isExpandedIndex} = this.state;
    const dialogStatus = statuses.filter(status => status.id === isExpandedIndex);

    const { className, 'data-cy': cy } = this.props;

    return (
      <section
       className={`tile-handler ${className}`} 
       onClick={this.onClick} 
       data-cy={`tile-handler-${cy}`}>
        {statuses.map(
          (status) => <Tile key={status.id} {...status} />
        )}
        {/* {this.state.isExpanded ? 
          <HealthCheck { ...{ isExpanded, ...dialogStatus[0] }} handleClose={this.handleClose}/>
          : null
        } */}
      </section>
    );
  }
}