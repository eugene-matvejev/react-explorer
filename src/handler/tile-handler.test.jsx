import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TileHandler from './tile-handler';

configure({ adapter: new Adapter() });

describe('<TileHandler/>', () => {
    const props = {
        onFilter: jest.fn((v) => v),
        data: [
            {
                id: 1,
            },
        ],
    };

    describe('render', () => {
        it('with default/require d props', () => {
            const c = shallow(<TileHandler {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['className', '{{className}}'],
                ['data-cy', '{{data-cy}}'], /** should be passed 'as is' */
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<TileHandler {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });

    describe('internal callbacks', () => {
        describe('::onChange', () => {
            it('should be invoked from a change on a [data-cy="-pattern"]', () => {
                const spy = spyOn(TileHandler.prototype, 'onChange');

                shallow(<TileHandler {...props} />)
                    .find('[data-cy="-pattern"]')
                    .simulate('change', { target: { value: 'v' } });

                expect(spy).toBeCalled();
            });

            it('should invoke external callback ::onChange', () => {
                const spy = jest.fn(({ data: v }) => v);

                shallow(<TileHandler {...props} onFilter={spy} />)
                    .find('[data-cy="-pattern"]')
                    .simulate('change', { target: { value: 'v' } });

                expect(spy).toBeCalled();
            });
        });
    });
});
