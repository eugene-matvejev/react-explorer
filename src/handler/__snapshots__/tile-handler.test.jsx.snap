// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileHandler/> render with default/require d props 1`] = `
<section
  className="tile-handler "
>
  <ComposedInput
    className=""
    data-cy="-pattern"
    label="filter"
    onChange={[Function]}
    placeholder="type pattern to search..."
  />
  <div
    className="tile-handler__container"
  >
    <Tile
      className=""
      data-cy="-tile-0"
      id={1}
      key="0"
      target="_blank"
      to="/explore/1"
    />
    <Tile
      className="tile__add-control"
      data-cy="-add"
      target="_blank"
      to="/new"
    />
  </div>
</section>
`;

exports[`<TileHandler/> render with optional props [::className] as "{{className}}" 1`] = `
<section
  className="tile-handler {{className}}"
>
  <ComposedInput
    className=""
    data-cy="-pattern"
    label="filter"
    onChange={[Function]}
    placeholder="type pattern to search..."
  />
  <div
    className="tile-handler__container"
  >
    <Tile
      className=""
      data-cy="-tile-0"
      id={1}
      key="0"
      target="_blank"
      to="/explore/1"
    />
    <Tile
      className="tile__add-control"
      data-cy="-add"
      target="_blank"
      to="/new"
    />
  </div>
</section>
`;

exports[`<TileHandler/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<section
  className="tile-handler "
>
  <ComposedInput
    className=""
    data-cy="{{data-cy}}-pattern"
    label="filter"
    onChange={[Function]}
    placeholder="type pattern to search..."
  />
  <div
    className="tile-handler__container"
  >
    <Tile
      className=""
      data-cy="{{data-cy}}-tile-0"
      id={1}
      key="0"
      target="_blank"
      to="/explore/1"
    />
    <Tile
      className="tile__add-control"
      data-cy="{{data-cy}}-add"
      target="_blank"
      to="/new"
    />
  </div>
</section>
`;
